gazebo-web/gazebosim-web-frontend
# Add /not-found page

Part of #16 (the easy part :sweat_smile: Bear with me as I learn about how the website works :sweat_smile: )

This creates the `/not-found` page, which is referred to twice by the doc page `doc.component.ts` [here](https://github.com/gazebo-web/gazebosim-web-frontend/blob/4da2478c9ed82db0ae0c3f21bb70627d6a7a91ea/src/app/doc/doc.component.ts#L207) and [here](https://github.com/gazebo-web/gazebosim-web-frontend/blob/4da2478c9ed82db0ae0c3f21bb70627d6a7a91ea/src/app/doc/doc.component.ts#L234)

This one-line page probably doesn't need a big CSS, but... I just copied from `support.component.scss`,which is an equally empty page :woman_shrugging: 

## To test

I used the `docker_dev` branch in both gazebosim-web-frontend and gazebosim-web-backend repos to test it (merge to that branch locally):
```
cd path/to/gazeobsim-web-frontend
docker build . -t gazebosim_web_frontend
cd path/to/gazebosim-web-backend
docker build . -t gazebosim_web_backend
cd path/to/gazeobsim-web-frontend
docker-compose up
```

Navigate to http://localhost:3000/not-found , and it should say "The page is not found." (as opposed to just showing the home page, which is the old behavior)
A doc URL with the distro name e.g. http://localhost:3000/docs/harmonic/bad_url should redirect to the `not-found` page (this is the second part of #16).

One small annoyance is that it appears to be a redirect, so if you hit the back button in the browser, you're just redirected back to the same page. You have to go back 2 pages to get back to the previous page.

(This PR does not address the first part of #16, which is for URLs without the distro name, lke http://localhost:3000/docs/bad_url )

I can't assign reviewers so tagging manually @azeey
